### PR TITLE
Added goreleaser changelog groups and update PR template

### PR DESCRIPTION
**Pull request by cpheps**
## Description of Changes
- Updated goreleaser with changelog groups
- Updated PR template to include instructions and remove changelog checkbox.

## **Please check that the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Add a changelog entry (for non-trivial bug fixes / features)
- [ ] CI passes

### CODE REVIEW

**Review by BinaryFissionGames — APPROVED** [review]
What's our plan with the changelog file? Seems redundant.

**Comment by cpheps** [comment]
> What's our plan with the changelog file? Seems redundant.

I agree. Probably add a big deprecated note at the top but keep it to preserve history.

**Review by BinaryFissionGames — DISMISSED** [review]

**Comment by jsirianni** [comment]
I forgot to mention, Dependabot can be told how it should name and label pull requests https://github.com/observIQ/iris/pull/209/files

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,7 +1,14 @@
+## Important (read before submitting)
+In order for changes to be captured in changelog correctly please add one of the following prefixes to the title. **Note** the parenthesis are optional and so is any text in them.
+- `feat(OPTIONAL):` = New features
+- `fix(OPTIONAL):` = Bug fixes
+- `deps(OPTIONAL):` = Dependency updates, primarily dependabot
+
+_Delete the above paragraph before submitting._
```
[the remark this answers is not in the document]
Comment (cpheps) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: Yeah I'll do that.

Anything you want can go in the parenthesis or just ignore them all together.

**Comment by cpheps** [comment]
> I forgot to mention, Dependabot can be told how it should name and label pull requests https://github.com/observIQ/iris/pull/209/files

I didn't know that I'll update.

**Review by Mrod1598 — APPROVED** [review]

**Review by BinaryFissionGames — APPROVED** [review]

**Comment by cpheps** [comment]
CI failures are unrelated to changes going to merge.

**Pull request merged**